src/arch/xtensa/lib/cpu.c

```diff
@@ -46,17 +71,23 @@ void arch_cpu_enable_core(int id)
 	irq_local_disable(flags);
 
 	if (!arch_cpu_is_core_enabled(id)) {
-		pm_runtime_get(PM_RUNTIME_DSP, id);
-
 		/* Turn on stack memory for core */
 		pm_runtime_get(CORE_MEMORY_POW, id);
 
+		pm_runtime_get(PM_RUNTIME_DSP, id);
```

Comment:
This looks like it's solving another problem and should be a new patch.

Reply:
ok, new commit added

Reply:
Can you add an inline comment here too, so it obvious we turn on core after the memory.

Reply:
Not seeing this yet, commit message could do with the reason "why" too.

Reply:
done now